php/blocks/class-rest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * REST API handling.
+ *
+ * @package   Block_Lab
+ * @copyright Copyright(c) 2020, Block Lab
+ * @license http://opensource.org/licenses/GPL-2.0 GNU General Public License, version 2 (GPL-2.0)
+ */
+
+namespace Block_Lab\Blocks;
+
+use Block_Lab\Component_Abstract;
+
+/**
+ * Class Block
+ */
+class Rest extends Component_Abstract {
+
+	/**
+	 * Register all the hooks.
+	 */
+	public function register_hooks() {
+		add_filter( 'rest_endpoints', [ $this, 'filter_block_endpoints' ] );
+	}
+
+	/**
+	 * Filters the Block Lab endpoints to allow POST requests.
+	 *
+	 * @param array $endpoints The REST API endpoints, an associative array of $route => $handlers.
+	 * @return array The filtered endpoints, with the Block Lab endpoints allowing POST requests.
+	 */
+	public function filter_block_endpoints( $endpoints ) {
+		foreach ( $endpoints as $route => $handler ) {
+			if ( 0 === strpos( $route, '/wp/v2/block-renderer/(?P<name>block-lab/' ) && isset( $endpoints[ $route ][0] ) ) {
+				$endpoints[ $route ][0]['methods']  = [ \WP_REST_Server::READABLE, \WP_REST_Server::CREATABLE ];
+				$endpoints[ $route ][0]['callback'] = [ $this, 'get_item' ];
+			}
+		}
+
+		return $endpoints;
+	}
+
+	/**
+	 * Returns block output from block's registered render_callback.
+	 *
+	 * Forked from WP_REST_Block_Renderer_Controller::get_item(), with a simple change to process POST requests.
+	 *
+	 * @todo: revert if this has been merged and enough version of Core have passed: https://github.com/WordPress/wordpress-develop/pull/196/
+	 * @see https://github.com/WordPress/wordpress-develop/blob/dfa959bbd58f13b504e269aad45412a85f74e491/src/wp-includes/rest-api/endpoints/class-wp-rest-block-renderer-controller.php#L121
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
```

Comment:
This method shouldn't be needed, per the feedback in the Core patch: https://core.trac.wordpress.org/ticket/49680#comment:11

`filter_block_endpoints()` will probably still be needed for years, though.